libs/contract/src/lib/negotiation/utils.ts

```diff
@@ -0,0 +1,110 @@
+import { LanguageRecord } from "@blockframes/movie/+state/movie.firestore";
+import { Language, Media, Territory } from "@blockframes/utils/static-model";
+import { BucketTerm, Duration } from "../term/+state";
+import { Negotiation, NegotiationStatus } from "./+state/negotiation.firestore";
+
+interface ArrayDifferences<T = string> {
+  added: T[];
+  removed: T[];
+}
+
+interface Difference<T> {
+  before: T,
+  after: T,
+  delta?: number;
+}
+
+interface TermDifference {
+  territories: ArrayDifferences<Territory>,
+  media: ArrayDifferences<Media>,
+  exclusivity: Difference<boolean>,
+  duration: DurationDiff,
+  languages: ArrayDifferences<LanguageRecord>,
+}
+
+interface NegotiationDifferences {
+  buyerId: Difference<string>,
+  price: Difference<number>,
+  createdByOrg: Difference<string>,
+  orgId: Difference<string>,
+  parentTermId: Difference<string>,
+  sellerId: Difference<string>,
+  specificity: Difference<string>,
+  stakeholders: ArrayDifferences<string>,
+  status: Difference<NegotiationStatus>,
+  titleId: Difference<string>,
+  terms: TermDifference,
+}
```

Comment:
only price & terms can change
```suggestion
interface NegotiationDifferences {
  price: Difference<number>,
  territories: ArrayDifferences<Territory>,
  media: ArrayDifferences<Media>,
  exclusivity: Difference<boolean>,
  duration: DurationDiff,
  languages: ArrayDifferences<LanguageRecord>,
}
```